### PR TITLE
FIX-#6199: make sure `read_html` return a list of DataFrames

### LINKED ISSUE #6199: `read_html` should return a list of DataFrames

### PR DESCRIPTION
**Pull request by anmyachev**
## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #6199 
- [ ] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

File: modin/core/io/io.py [hunk 1]
```diff
@@ -238,26 +238,27 @@ def read_html(
...
+            encoding=encoding,
+            decimal=decimal,
+            converters=converters,
+            na_values=na_values,
+            keep_default_na=keep_default_na,
+            displayed_only=displayed_only,
+            **kwargs,
         )
+        if isinstance(result, (pandas.DataFrame, pandas.Series)):
+            return (cls.from_pandas(result),)
```
Comment (dchigarev) [inline modin/core/io/io.py:260]: pandas always return a list of dfs, so does this case ever happen?

Comment (anmyachev) [reply modin/core/io/io.py:260]: good point!

Comment (anmyachev) [inline modin/core/io/io.py:260]:
```suggestion
```

Comment (dchigarev) [reply modin/core/io/io.py:260]: the logic is still here, did you figure out a case when this happens?

Comment (anmyachev) [reply modin/core/io/io.py:260]: It came back due to a force push.

Comment (anmyachev) [reply modin/core/io/io.py]: @dchigarev fixed

**Review by dchigarev — APPROVED** [review]

**Pull request merged**